persistent-sqlite/Database/Persist/Sqlite.hs

```diff
@@ -484,19 +481,29 @@ escape (DBName s) =
 
 -- | Information required to setup a connection pool.
 data SqliteConf = SqliteConf
+    { sqlDatabase :: Text
+    , sqlPoolSize :: Int
+    }
+    | SqliteConfInfo
     { sqlConnInfo :: SqliteConnectionInfo
     , sqlPoolSize :: Int
     } deriving Show
 
 instance FromJSON SqliteConf where
-    parseJSON v = modifyFailure ("Persistent: error loading Sqlite conf: " ++) $
-      flip (withObject "SqliteConf") v $ \o -> SqliteConf
-        A.<$> o .: "database"
-        A.<*> o .: "poolsize"
+    parseJSON v = modifyFailure ("Persistent: error loading Sqlite conf: " ++) $ flip (withObject "SqliteConf") v parser where
+        parser o = if HashMap.member "database" o
```

Comment:
This works fine, but for future reference the `Alternative` instance would probably be easier:

```haskell
(SqliteConf <$>o .: "database" ...) <|> (SqliteConfInfo <$> o.: "connInfo" ...)
```